src/audio/google_rtc_audio_processing_mock.c

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Google LLC.
+//
+// Author: Lionel Koenig <lionelk@google.com>
+#include "google_rtc_audio_processing.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "sof/lib/alloc.h"
+#include "ipc/topology.h"
+
+#define GOOGLE_RTC_AUDIO_PROCESSING_SAMPLE_RATE_HZ 48000
+
+struct GoogleRtcAudioProcessingState {
+	int num_capture_channels;
+	int num_aec_reference_channels;
+	int num_output_channels;
+	int num_frames;
+	int16_t *aec_reference;
+};
+
+GoogleRtcAudioProcessingState *GoogleRtcAudioProcessingCreate()
+{
+	struct GoogleRtcAudioProcessingState *s =
+			rballoc(0, SOF_MEM_CAPS_RAM, sizeof(GoogleRtcAudioProcessingState));
+	s->num_capture_channels = 1;
+	s->num_aec_reference_channels = 1;
+	s->num_output_channels = 1;
+	s->num_frames = GOOGLE_RTC_AUDIO_PROCESSING_SAMPLE_RATE_HZ * 10 / 1000;
+	s->aec_reference = rballoc(0,
+							   SOF_MEM_CAPS_RAM,
```

Comment:
Indentation looks wrong here, could be github though ?

Reply:
Unfortunately I haven't found a way to get this one right. I am open to any suggestion.